Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -4306,4 +4306,108 @@ final class HTTPClientTests: XCTestCaseHTTPClientTestsBaseClass {
+            )
+        )
+        defer { XCTAssertNoThrow(client.shutdown()) }
+
+        let bin = HTTPBin(.http1_1(ssl: true, compress: false))
+        defer { XCTAssertNoThrow(try bin.shutdown()) }
+
+        for _ in 0..<3 {
+            XCTAssertNoThrow(try client.get(url: "https://localhost:\(bin.port)/get").wait())
+        }
+
+        // Even though multiple requests were made, the connection debug initializer must be called
+        // only once.
+        XCTAssertEqual(connectionDebugInitializerUtil.executionCount.withLockedValue { $0 }, 1)
+    }
+
+    func testHTTP2ConnectionAndStreamChannelDebugInitializers() {
+        let connectionDebugInitializerUtil = CountingDebugInitializerUtil()
+        let streamChannelDebugInitializerUtil = CountingDebugInitializerUtil()
+
+        // Initializing even with just `http2ConnectionDebugInitializer` and
+        // `http2StreamChannelDebugInitializer` (rather than manually modifying `config`) to ensure
+        // that the matching `init` actually wires up these arguments with the respective
+        // properties. This is necessary as these parameters are defaulted and can be easy to miss.
+        var config = HTTPClient.Configuration(
+            http2ConnectionDebugInitializer:
+                connectionDebugInitializerUtil.initialize(channel:),
+            http2StreamChannelDebugInitializer:
+                streamChannelDebugInitializerUtil.initialize(channel:)
+        )
+        config.tlsConfiguration = .clientDefault
+        config.tlsConfiguration?.certificateVerification = .none
+        config.httpVersion = .automatic
+
+        let client = HTTPClient(
+            eventLoopGroupProvider: .singleton,
+            configuration: config,
+            backgroundActivityLogger: Logger(
+                label: "HTTPClient",
+                factory: StreamLogHandler.standardOutput(label:)
+            )
+        )
+        defer { XCTAssertNoThrow(client.shutdown()) }
+
+        let bin = HTTPBin(.http2(compress: false))
+        defer { XCTAssertNoThrow(try bin.shutdown()) }
+
+        let numberOfRequests = 3
+
+        for _ in 0..<numberOfRequests {
+            XCTAssertNoThrow(try client.get(url: "https://localhost:\(bin.port)/get").wait())
+        }
+
+        // Even though multiple requests were made, the connection debug initializer must be called
+        // only once.
+        XCTAssertEqual(
+            connectionDebugInitializerUtil.executionCount.withLockedValue { $0 },
+            1
+        )
+
+        // The stream channel debug initializer must be called only as much as the number of
+        // requests made.
+        XCTAssertEqual(
+            streamChannelDebugInitializerUtil.executionCount.withLockedValue { $0 },
+            numberOfRequests
+        )
+    }
+}
+
+final class CountingDebugInitializerUtil: Sendable {
+    let executionCount: NIOLockedValueBox<Int>
+
+    /// The acual debug initializer.
+    @Sendable
```

Comment:
I don't think we need this annotation as the type is `Sendable`

Reply:
I think that's because you do `http1_1ConnectionDebugInitializer: connectionDebugInitializerUtil.initialize(channel:)` rather than:

```swift
http1_1ConnectionDebugInitializer: { channel in 
   connectionDebugInitializerUtil.initialize(channel: channel) 
}
```